### DISCUSSION TITLE
KBR Add Corrections –  010groopsInstruments.xml – "GRACE gravity field recovery” Scenario

### DISCUSSION THREAD

**Discussion by grkal** [groops-devs/groops#289] [Q&A] [answered]
Dear GROOPS Team,

The topic of this message concerns the “InstrumentMultiplyAdd” program with the comment “KBR add corrections” included in the “GroupPrograms” program with the comment “KBR” of the “010groopsInstruments.xml” configuration file provided with the “GRACE gravity field recovery” scenario.

In this program:
1.	the light time correction data file:

- satellite/{timeStart:%y-%m}\_shortArc/gracefo_lightTime\_{timeStart:%y-%m}.dat

and

2.	the antenna offset correction data file:

- satellite/{timeStart:%y-%m}\_shortArc/gracefo_antCentr\_{timeStart:%y-%m}.dat

are added to the simulated satellite tracking data file:

- {tmpDir}/gracefo\_satelliteTrackingSim\_{timeStart:%y-%m}.dat

as depicted in the image below:

![20241009_KBRaddCorrections010groopsInstruments](https://github.com/user-attachments/assets/3e2a3e98-f766-4fdf-a895-3f05b31de4fb)

Shouldn’t these two corrections be added to the real satellite tracking data file, i.e.:

- satellite/{timeStart:%y-%m}\_shortArc/gracefo_satelliteTracking\_{timeStart:%y-%m}.dat,

instead of the simulated one as also described in the “GRACE-FO Level-1 Data Product User Handbook JPL D-56935” (Wen, et al., 2017) in section “4.2.11 KBR”? According to my understanding, the generated simulated data file should not need these corrections.

Thank you very much in advance for your time and looking forward to your feedback.

Best regards,
Grigorios Kalimeris
Chair of Astronomical and Physical Geodesy
Technical University of Munich

Bibliography:

Wen, H. Y., Kruizinga, G., Paik, M., Landerer, F., Bertiger, W., Sakumura, C., Bandikova, T., Mccullough, C. M. (2019, September 11). Gravity Recovery and Climate Experiment Follow-On (GRACE-FO) level-1 data product user handbook JPL D-56935. NASA Jet Propulsion Laboratory - California Institute of Technology. Retrieved on October 9, 2024, from PO.DAAC / JPL / NASA GRACE-FO Mission Documentation Website: https://podaac.jpl.nasa.gov/gravity/gracefo-documentation

**Comment by felingeroehlix** [comment, marked as answer]
Dear Grigorios,

Thanks a lot for your remark. You are completely right that the light time correction and the antenna offset correction have to be added to the actual satellite tracking data. This is also what is done in the gravity field scenario whenever satellite tracking data is required:
- PreprocessingVariationalEquationSstFit (020groopsVariational.xml)
- PreprocessingSst (030groopsPreprocessing.xml)
- NormalsBuildShortTimeStaticLongTime (040groopsMonthlyNormals.xml)

In 010groopsInstruments.xml simulated satellite tracking data is used for a coarse outlier detection using a threshold. However, you are right that the two corrections are wrongfully added to the simulated data. Since in InstrumentRemoveEpochsByCriteria the satellite tracking data (without the corrections) is validated against the simulated data, the corrections should either be subtracted from the simulated data or added to the actual satellite tracking data. 

![Screenshot at 2024-10-16 13-29-42](https://github.com/user-attachments/assets/71cb7564-e55a-4d24-90b8-ec5a59979f4f)

So, in conclusion the factor of the light time correction and the antenna center correction (InstrumentMultiplyAdd in 010groopsInstruments.xml) should be -1 not 1^^. I also changed the file in the scenario folder.

Thanks for your contribution,
Felix

**Reply by grkal** [reply]
Dear Felix,

Thank you very much for your detailed reply.

Best regards,
Greg
